lib/cisco_node_utils/radius_server.rb

```diff
@@ -48,24 +69,65 @@ def self.radiusservers
       radiusservers_list = config_get('radius_server', 'hosts')
       return hash if radiusservers_list.empty?
       radiusservers_list.each do |id|
-        hash[id] = RadiusServer.new(id, false)
+        if platform == :ios_xr
+          authp = config_get('radius_server', 'auth-port', ip: id)
+          authp = authp[0] if authp.is_a?(Array)
+          authp = authp.to_i
+
+          acctp = config_get('radius_server', 'acct-port', ip: id)
+          acctp = acctp[0] if acctp.is_a?(Array)
+          acctp = acctp.to_i
+
+          hash[id] = RadiusServer.new(id, false, authp, acctp)
+        else
+          hash[id] = RadiusServer.new(id, false)
+        end
       end
 
       hash
     end
 
     def create
+      destroy if platform == :ios_xr
       config_set('radius_server',
                  'hosts',
-                 state: '',
-                 ip:    @name)
+                 state:     '',
+                 ip:        @name,
+                 auth_port: @auth_port,
+                 acct_port: @acct_port)
     end
 
     def destroy
-      config_set('radius_server',
-                 'hosts',
-                 state: 'no',
-                 ip:    @name)
+      if platform == :ios_xr
+        # This provider only support a 1-1 mapping between host and VRF.
+        # Thus, we must remove the other entries on different VRFs.
+        all_hosts = config_get('radius_server', 'host_port_pairs', ip: @name)
+        return unless all_hosts.is_a?(Array)
+
+        warn("#{name} is configured multiple times on the device" \
```

Comment:
I'd recommend adding a test case to the minitest that explicitly exercises this functionality (and also verifies the behavior of the various getters/setters in the case where multiple entries exist).